### PR TITLE
fix/33797: Conversation - Back arrow leads to LHN instead of group lists after page refresh in a profile screen

### PR DESCRIPTION
**Pull request by Tony-MK**
### Fixed Issues

$ https://github.com/Expensify/App/issues/33797
PROPOSAL: https://github.com/Expensify/App/issues/33797#issuecomment-1872583750

### Tests

1. Go to the application and log in.
2. Create a group chat, if none exists.
3. Go to the group chat.
4. Open the group details > members list.
5. Visit a member's profile page and refresh it.
6. Verify that the same member's profile page is displayed like in step 5.
7. Tap on the back button.
8. Verify the app navigated back to the member's list mentioned in step 4.

- [x] Verify that no errors appear in the JS console

### Offline tests

1. Go to the application and log in.
2. Create a group chat, if none exists.
3. Go to the group chat.
4. Open the group details > members list.
5. Visit a member's profile page and refresh it.
6. Verify that the same member's profile page is displayed like in step 5.
7. Tap on the back button.
8. Verify the app navigated back to the member's list mentioned in step 4.

### QA Steps

1. Go to the application and log in.
2. Create a group chat, if none exists.
3. Go to the group chat.
4. Open the group details > members list.
5. Visit a member's profile page and refresh it.
6. Verify that the same member's profile page is displayed like in step 5.
7. Tap on the back button.
8. Verify the app navigated back to the member's list mentioned in step 4.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/22982173/c343843e-e5d3-46f4-8bbc-337cc71d44d5

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/22982173/e0acfa94-7668-4139-9a04-ff285caddc95
</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/22982173/a3b181c7-bd6d-46e8-a8f2-916b92e1c54a
</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/22982173/6305a737-ecf0-402a-8431-19fd410ff447
</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/22982173/33fbf011-ef74-49c0-9c56-2779c53c8b5e
</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/22982173/5955bf1a-3952-4219-bfb4-8cea2762070d
</details>

### CODE REVIEW

**Comment by Tony-MK** [comment]
@abdulrahuman5196, After reloading the profile page, is it okay if the app navigates to the report instead of the details page when it is on the participant's page?

If the app is not expected to behave like the reproduction video below, I can investigate again and try to fix it. 

https://github.com/Expensify/App/assets/22982173/dc175403-a90f-4b92-9e16-4e7c05a5d55b

**Comment by abdulrahuman5196** [comment]
@Tony-MK 
> Hey @abdulrahuman5196, after reloading the ProfilePage and navigating back to the ReportParticipantsPage, is the app expected to navigate to the ReportDetailsPagedirectly to the report instead of?

Yes. Would be good to fix the same.

And kindly fix other ingress for the profile screen as well as pointed out by other contributors in the GH.

**Comment by Tony-MK** [comment]
@abdulrahuman5196, could you Kindly review the PR? 

Furthermore, the `NotFoundPage` appears after reloading when testing with `iOS - mWeb` and `macOS - Desktop`.

However, it navigates back to the LHN as expected.

Is that going to be a problem? 🤔 

Thank you.

File: src/pages/ReportParticipantsPage.js [hunk 1]
```diff
@@ -121,7 +122,7 @@ function ReportParticipantsPage(props) {
                                     },
                                 ]}
                                 onSelectRow={(option) => {
-                                    Navigation.navigate(ROUTES.PROFILE.getRoute(option.accountID));
```
Comment (Tony-MK) [inline src/pages/ReportParticipantsPage.js:124]: Do you think it's good to add comments? 🤔

File: src/pages/ReportParticipantsPage.js [hunk 2]
```diff
@@ -121,7 +122,7 @@ function ReportParticipantsPage(props) {
...
+                                    Navigation.navigate(ROUTES.PROFILE.getRoute(option.accountID, ROUTES.REPORT_PARTICIPANTS.getRoute(props.report.reportID)));
```
Comment (Tony-MK) [inline src/pages/ReportParticipantsPage.js:125]: Should I add a comment? 🤔

**Comment by abdulrahuman5196** [comment]
thank you for the changes. Will go through the PR today or tomorrow morning.

**Comment by abdulrahuman5196** [comment]
> @Tony-MK
> 
> > Hey @abdulrahuman5196, after reloading the ProfilePage and navigating back to the ReportParticipantsPage, is the app expected to navigate to the ReportDetailsPagedirectly to the report instead of?
> 
> Yes. Would be good to fix the same.
> 
> And kindly fix other ingress for the profile screen as well as pointed out by other contributors in the GH.

@Tony-MK Did you check on this? There are other ingress for the same page right? That should also be affected by this issue right?

**Comment by Tony-MK** [comment]
> @Tony-MK Did you check on this? There are other ingresses for the same page right? That should also be affected by this issue right?

@abdulrahuman5196 Yeah, I checked on the other ingress mentioned by the other contributors and made the changes respectively.

**Comment by Tony-MK** [comment]
Hi, @abdulrahuman5196, just a gentle reminder. 

Could we get this PR merged this week?

Thank you.

**Comment by abdulrahuman5196** [comment]
Sorry for the delay. Let me take a look into this today,

**Comment by abdulrahuman5196** [comment]
Reviewing now

**Comment by abdulrahuman5196** [comment]
Initial testing looks good. Working on filling the checklist.

**Comment by abdulrahuman5196** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>




https://github.com/Expensify/App/assets/46707890/1f895130-7172-4f4f-b72d-52412e179d78

</details>

<details>
<summary>Android: mWeb Chrome</summary>




https://github.com/Expensify/App/assets/46707890/44d5c6e9-db6b-4233-b068-f275fb01ffc4




</details>

<details>
<summary>iOS: Native</summary>





https://github.com/Expensify/App/assets/46707890/226bcb1b-48ce-4c0c-a34a-6731f621f883




</details>

<details>
<summary>iOS: mWeb Safari</summary>





https://github.com/Expensify/App/assets/46707890/9b84166d-37f6-4077-b4e6-88ad9590f9e1




</details>

<details>
<summary>MacOS: Chrome / Safari</summary>





https://github.com/Expensify/App/assets/46707890/2bb1cbad-d0fe-4117-93d9-d5e49bb73ad6



</details>

<details>
<summary>MacOS: Desktop</summary>




https://github.com/Expensify/App/assets/46707890/10119e77-eb32-4ddd-9fc2-bb7758111159




</details>

**Review by abdulrahuman5196 — APPROVED** [review]
Changes looks good and works well. Reviewers checklist is also complete.

All yours. @Gonals 

🎀 👀 🎀
C+ Reviewed

**Review by Gonals — APPROVED** [review]

**Pull request merged**